### DISCUSSION TITLE
No ideal way to search for the word under the cursor and remain stationary.

### DISCUSSION THREAD

**Discussion by otakutyrant** [neovim/neovim#24285] [General]
See [Vim: search and highlight but do not jump](https://stackoverflow.com/questions/4256697/vim-search-and-highlight-but-do-not-jump). In past, I do not want `*` jump immediately so I installed a plugin to disable it. Many years passed, now I wonder there is still no official solution available.

In restrospect, the document of `*` says: Search forward for the [count]'th occurrence of the word nearest to the cursor. So I tried `0*` to expect search for the word under the cursor and remain stationary. By instinct, zeroth occurence comes before before the first one, and that is exactly the word itself under the cursor. Unfortunately, it still jumps, behaving like `1*` or `*`.

**Comment by justinmk** [comment]
You don't need  a plugin, just use this mapping:

    nnoremap <silent> *  ms:<c-u>let @/='\V\<'.escape(expand('<cword>'), '/\').'\>'<bar>call histadd('/',@/)<bar>set hlsearch<cr>

**Reply by XXiaoA** [reply]
I am a kinda confused about the function of `ms`. Why we should create a mark `s` here. Do we use it later?

**Reply by nyngwang** [reply]
Not important but: is there a lua (`keymap.set` function) version? Too old to read Vim Script.

**Reply by XXiaoA** [reply]
Here is my method. But I prefer to viml
```lua
local fn = vim.fn
vim.keymap.set("n", "*", function()
    fn.setreg("/", [[\V\<]] .. fn.escape(fn.expand("<cword>"), [[/\]]) .. [[\>]])
    fn.histadd("/", fn.getreg("/"))
    vim.o.hlsearch = true
end)
```

**Reply by nyngwang** [reply]
@XXiaoA Thanks :) Another question: is it possible to make `*` don't jump to the next occurrence when I use it at the word under cursor?

**Reply by XXiaoA** [reply]
Have really open nvim and do some try? the above keymap is aimed to do this.

**Reply by nyngwang** [reply]
@XXiaoA You're right. Thank you! (I updated the config locally and I was still using the remote ssh Ubuntu where I didn't pull the changes :P)

**Reply by otakutyrant** [reply]
@XXiaoA Why do you use `vim.o` instead of `vim.opt`?

**Reply by otakutyrant** [reply]
@justinmk Truly excellent! You transcended all answers from Stack Overflow. Thank you.
